docs/core/diagnostics/distributed-tracing-concepts.md

```diff
@@ -0,0 +1,137 @@
+A will be restored as the current Activity on the thread. When an Activity is started it
+captures the current time as the
+<xref:System.Diagnostics.Activity.StartTimeUtc?displayProperty=nameWithType>. When it
+stops <xref:System.Diagnostics.Activity.Duration?displayProperty=nameWithType> is calculated
+as the difference between the current time and the start time.
+
+### Coordinating across process boundaries
+
+In order to track work across process boundaries Activity parent IDs need to be transmitted across
+the network so that the receiving process can create Activities that refer to them. When using
+W3C TraceContext ID format .NET will also use the HTTP headers recommended by
+[the standard](https://www.w3.org/TR/trace-context/) to transmit this information. When using the
+<xref:System.Diagnostics.ActivityIdFormat.Hierarchical> ID format
+.NET uses a custom request-id HTTP header to transmit the ID. Unlike many other language runtimes
+.NET in-box libraries natively understand how to decode and encode Activity IDs on HTTP messages
+as well as how to flow the ID through sychronous and asynchronous calls. This means that .NET
+applications that receive and emit HTTP messages participate in flowing distributed trace IDs
+automatically, with no special coding by the app developer nor 3rd party library dependencies.
+3rd party libraries may add support for transmitting IDs over non-HTTP message protocols or
+supporting custom encoding conventions for HTTP.
+
+### Collecting traces
+
+Instrumented code can create <xref:System.Diagnostics.Activity> objects
+as part of a distributed trace, but the information in these objects needs to be transmitted
+and serialized in a centralized persistant store so that the entire trace can be usefully reviewed
+later. There are several telemetry collection libraries that can do this task such as
+[Application Insights](https://docs.microsoft.com/azure/azure-monitor/app/distributed-tracing),
+[OpenTelemetry](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/getting-started/README.md),
+or a library provided by a 3rd party telemetry or APM vendor. Alternately developers can author
+their own custom Activity telemetry collection by using
+<xref:System.Diagnostics.ActivityListener?displayProperty=nameWithType> or
+<xref:System.Diagnostics.DiagnosticListener?displayProperty=nameWithType>. ActivityListener
+supports observing any Activity regardless whether the developer has any a-priori knowledge about it.
+This makes ActivityListener a simple and flexible general purpose solution. By contrast using
+DiagnosticListener is a more complex scenario that requires the instrumented code to opt-in by
+invoking <xref:System.Diagnostics.DiagnosticSource.StartActivity%2A?displayProperty=nameWithType> and
+the collection library needs to know the exact naming information that the instrumented code
+used when starting it. Using DiagnosticSource and DiagnosticListener allows the creator
```

Comment:
We could state that the in-box libraries publish DiagnosticSource events to pass things like the HttpContext instances, which may be leveraged by listeners to extract additional information to enrich the Activity.

Reply:
I think I will avoid going to into that level of detail here. I agree there is lots more we could say about DiagnosticSource which could probably could fill its own pages.